src/soundsourcemp3.cpp

```diff
@@ -515,6 +478,10 @@ unsigned SoundSourceMp3::read(unsigned long samples_wanted, const SAMPLE * _dest
                 break;
             }
         }
+        if (frameBefore == Stream->this_frame) {
+            // No seek, try again
+            continue;
```

Comment:
This seems to happen in the following situations
* mad_frame_decode != 0
* mad_frame_decode == 0: At the beginning of the stream, i.e. for the first frame